rust/connlib/libs/tunnel/src/tun_android.rs

```diff
@@ -44,18 +75,52 @@ impl IfaceDevice {
     }
 
     pub fn set_non_blocking(self) -> Result<Self> {
-        // Anrdoid already opens the tun device in non-blocking mode for us
+        // Android already opens the tun device in non-blocking mode and we can't change it from
+        // here.
         log::debug!("`set_non_blocking` unimplemented on Android");
         Ok(self)
     }
 
+    pub fn name(&self) -> Result<String> {
+        let mut ifr = ifreq {
+            ifr_name: [0; IFNAMSIZ],
+            ifr_ifru: unsafe { std::mem::zeroed() },
+        };
```

Comment:
Long term, I'd recommend to separate these FFI calls out into dedicated functions. You want the layer that uses `unsafe` to be as thin as possible and build safe abstractions over it.

Reply:
Noted, will see how @conectado wants to refactor this for better long-term maintainability.

Reply:
I think this is the level of abstraction for the unsafe to be, this is just a `ioctl` call and its preparation, it feels like each unsafe call here requires knowledge of the previous one so it makes sense as it is.